server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -108,6 +112,51 @@ public void uploadBlobs(
 
     }
 
+    @Override
+    public void uploadBlobAsync(
+        InputStream inputStream,
+        Iterable<String> remotePath,
+        String blobName,
+        WritePriority writePriority,
+        ActionListener<Void> listener
+    ) throws IOException {
+        assert remotePath instanceof BlobPath;
+        BlobPath blobPath = (BlobPath) remotePath;
+        final BlobContainer blobContainer = blobStore.blobContainer(blobPath);
+        if (blobContainer instanceof AsyncMultiStreamBlobContainer == false) {
+            blobContainer.writeBlob(blobName, inputStream, inputStream.available(), false);
+            listener.onResponse(null);
+            return;
+        }
+        final String resourceDescription = "BlobStoreTransferService.uploadBlob(blob=\"" + blobName + "\")";
+        byte[] bytes = inputStream.readAllBytes();
```

Comment:
how big can be the contents in the input stream? if the size is big it is going to allocate a large byte array. We can introduce a buffer and reuse it to limit the memory allocation during the transfer.

Reply:
Metadata will split across multiple files like index metadata, settings, index templates. Each of the individual files should be less than 5 MB.